index.html

```diff
@@ -2531,72 +2581,76 @@ <h3><dfn>Writing content</dfn></h3>
-                reject |promise| with a {{TypeError}} and abort these steps.
-              </li>
-              <li>
-                If running the <a>validate local type</a> steps on |record|'s
-                <a>recordType</a> returns `false`, reject |promise| with a
-                {{TypeError}} and abort these steps.
-              </li>
-              <li>
-                Return the result of running
-                <a>map local type to NDEF</a> given |record| and |ndef|.
-              </li>
-            </ul>
-          <li>
-            Switching on |record|'s <a>recordType</a>, pass |record| and |ndef|
-            to one of the following algorithms and return the result.
-            If the algorithm throws an exception |e|, reject |promise| with |e|
+            Switching on |record|'s <a>recordType</a>, invoke the algorithm
+            specified below with |record|, |ndef| and |context| and return the
+            result. If an exception |e| is thrown, reject |promise| with |e|
             and abort these steps.
             <dl>
               <dt>"`empty`"</dt>
               <ul>
                 <li>
-                  Return <a>map empty record to NDEF</a> given |record| and |ndef|.
+                  <a>map empty record to NDEF</a>.
                 </li>
               </ul>
               <dt>"`text`"</dt>
               <ul>
                 <li>
-                  Return <a>map text to NDEF</a> given |record| and |ndef|.
+                  <a>map text to NDEF</a>.
                 </li>
               </ul>
               <dt>"`url`"</dt>
               <ul>
                 <li>
-                  Return <a>map a URL to NDEF</a> given |record| and |ndef|.
+                  <a>map a URL to NDEF</a>.
                 </li>
               </ul>
               <dt>"`mime`"</dt>
               <ul>
                 <li>
-                  Return <a>map binary data to NDEF</a> given |record| and |ndef|.
+                  <a>map binary data to NDEF</a>.
                 </li>
               </ul>
               <dt>"`smart-poster`"</dt>
               <ul>
                 <li>
-                  Return <a>map smart poster to NDEF</a> given |record| and
-                  |ndef|.
+                  <a>map smart poster to NDEF</a>.
                 </li>
               </ul>
               <dt>"`absolute-url`"</dt>
               <ul>
                 <li>
-                  Return <a>map absolute-URL to NDEF</a> given |record| and
-                  |ndef|.
+                  <a>map absolute-URL to NDEF</a>.
                 </li>
               </ul>
             </dl>
           </li>
+          <li>
+            If |record|'s <a>recordType</a> starts with a colon `U+003A` (`:`):
+            <ul>
+              <li>
+                If |record| is not a payload to another <a>NDEF record</a>
+                and if |context| is not `"smart-poster"`,
```

Comment:
@zolkis, looks like we don't need to check `if |context| is not "smart-poster"`.

Reply:
I know that smart-poster contains other records, this is more like a shortcut (not to run a full check on containment since we already know the context). But it should be the first condition then.

Reply:
Usually "not a payload to another NDEF record" should be a defined algorithm, actually this might be an issue. We may want to extend the usage of `context` to help that. But frankly I don't like very much these meta-arguments (just for the sake of algorithms).

Reply:
OK, make sense, thanks!

Reply:
> Usually "not a payload to another NDEF record" should be a defined algorithm, actually this might be an issue. We may want to extend the usage of `context` to help that. But frankly I don't like very much these meta-arguments (just for the sake of algorithms).

why not just create an issue for further improvement? 😄

Reply:
I think it would be enough to use 

> If |context| is `""` (i.e. |record| is not a payload to another  <a>NDEF record</a>), reject ...